utopia-remix/app/components/sharingDialog.tsx

```diff
@@ -133,69 +134,128 @@ function SharingDialog({ project }: { project: ProjectListing | null }) {
-        return
-      }
+    (tokenId: string) => {
       approveAccessRequestFetcher.submit(
-        operationApproveAccessRequest(project, tokenId),
+        operationApproveAccessRequest(projectId, tokenId),
         { tokenId: tokenId },
         {
           method: 'POST',
           action: `/internal/projects/${projectId}/access/request/${tokenId}/approve`,
         },
       )
     },
-    [approveAccessRequestFetcher, project],
+    [approveAccessRequestFetcher, projectId],
   )
 
+  const hasGonePrivate = React.useMemo(() => {
+    return accessRequests.requests.length > 0 && accessLevel === AccessLevel.PRIVATE
+  }, [accessLevel, accessRequests])
+
+  const showAccessRequests = React.useMemo(() => {
+    return accessRequests.state === 'ready' && accessRequests.requests.length > 0
+  }, [accessRequests])
+
   return (
     <AnimatePresence>
-      {when(
-        accessRequests.state === 'ready' && accessRequests.requests.length > 0,
-        <motion.div
-          animate={{ height: 'auto', opacity: 1 }}
-          initial={{ height: 0, opacity: 0 }}
-          exit={{ height: 0, opacity: 0 }}
-        >
-          <Flex direction={'column'} gap='4'>
-            <Separator size='4' />
-            <AccessRequests
-              project={project}
-              approveAccessRequest={approveAccessRequest}
-              accessRequests={accessRequests.requests}
-            />
-          </Flex>
-        </motion.div>,
-      )}
+      <motion.div>
+        <Flex direction={'column'} gap='4'>
+          <Separator size='4' />
+
+          {when(hasGonePrivate, <HasGonePrivate />)}
+
+          <OwnerCollaboratorRow />
+
+          {when(
+            showAccessRequests,
+            <motion.div
+              animate={{ height: 'auto', opacity: 1 }}
+              initial={{ height: 0, opacity: 0 }}
+              exit={{ height: 0, opacity: 0 }}
+            >
+              <Flex direction={'column'} gap='4'>
+                <AccessRequests
+                  projectId={projectId}
+                  projectAccessLevel={accessLevel}
+                  approveAccessRequest={approveAccessRequest}
+                  accessRequests={accessRequests.requests}
+                />
+              </Flex>
+            </motion.div>,
+          )}
+        </Flex>
+      </motion.div>
     </AnimatePresence>
   )
 })
 AccessRequestsList.displayName = 'AccessRequestsList'
 
+const HasGonePrivate = React.memo(() => {
+  return (
+    <Text size='1' style={{ opacity: 0.5 }}>
+      This project was changed to private, previous collaborators can no longer view it.
+    </Text>
+  )
+})
+HasGonePrivate.displayName = 'HasGonePrivate'
+
+const OwnerCollaboratorRow = React.memo(() => {
+  const myUser = useProjectsStore((store) => store.myUser)
+  if (myUser == null) {
+    return null
+  }
+
+  return (
+    <CollaboratorRow
+      picture={myUser?.picture ?? null}
+      name={`${myUser?.name ?? myUser?.email ?? myUser?.id} (you)`}
```

Comment:
Should this have a final default value? I think it can end up with `null (you)` right?

Reply:
nope, in the _worstest_ case it will be your user id

Reply:
(those `?` are actually incorrect since the not-null is checked above, will make a PR to clean up)